pkg/helm/watches/watches.go

```diff
@@ -15,59 +15,71 @@
 package watches
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
-
-	// todo(camila): replace it for yaml "sigs.k8s.io/yaml"
-	// See that the unmarshaling JSON will be affected
-	yaml "gopkg.in/yaml.v3"
+	"path/filepath"
+	"text/template"
 
 	"helm.sh/helm/v3/pkg/chartutil"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/yaml"
+
+	"github.com/operator-framework/operator-sdk/internal/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/helm"
+	"github.com/operator-framework/operator-sdk/internal/util/fileutil"
 )
 
+const WatchesFile = "watches.yaml"
+
 // Watch defines options for configuring a watch for a Helm-based
 // custom resource.
 type Watch struct {
-	GroupVersionKind        schema.GroupVersionKind `yaml:",inline"`
-	ChartDir                string                  `yaml:"chart"`
-	WatchDependentResources *bool                   `yaml:"watchDependentResources,omitempty"`
-	OverrideValues          map[string]string       `yaml:"overrideValues,omitempty"`
-}
-
-func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
This method should be deprecated instead of removed since it's user-facing.

Reply:
I don't think we're considering any of the Helm operator library code to be public (even though it _technically_ is).

@fabianvf WDYT? Do you consider the Ansible operator code public?

Reply:
hmm, not really, but at the same time it is there and if anyone is actually trying to make a hybrid operator removing stuff willy-nilly could be annoying. If it's not too much extra effort I think a deprecation -> removal approach is probably better, though I'm not sure how many people, if any, would be impacted at this point.

Reply:
It's better to get into the habit of both going through a deprecation process and being concious of what code we make public, although I'm inclined to make an exception in this particular case.

Reply:
I'm okay with that idea. What do you think of this function, in particular though? Is it reasonable to assume that hybrid operator developers would be using `yaml.Unmarshal` instead of calling this directly?

Reply:
Hard to say as it's a very hypothetical group, but they'd probably be using `yaml.Unmarshal`

Reply:
I'll deprecate this for now since it shouldn't hurt anything to keep it around.

Reply:
Done!